src/fedora/39/amd64/Dockerfile

```diff
@@ -0,0 +1,59 @@
+FROM fedora:39
+
+# Install the base toolchain we need to build anything (clang, cmake, make and the like)
+# this does not include libraries that we need to compile different projects, we'd like
+# them in a different layer.
+RUN dnf --setopt=install_weak_deps=False install -y \
+        clang \
+        cmake \
+        dnf-plugins-core \
+        findutils \
+        gdb \
+        glibc-langpack-en \
+        lldb-devel \
+        llvm-devel \
+        make \
+        python \
+        which \
+    && dnf clean all
+
+# Add MS package repo.
+COPY  microsoft.asc /tmp
+RUN dnf config-manager --add-repo https://packages.microsoft.com/fedora/37/prod/ && \
```

Comment:
Can this be `39`? I checked, and the repo does exist: https://packages.microsoft.com/fedora/39/

Reply:
for now, the binary is identical but it would be good to keep track. If we bump to cases when the repo is missing we should push it to MsQuic. 

cc: @CarnaViire

Reply:
Is there a reason why `fedora 38` uses `36`? https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/main/src/fedora/38/amd64/Dockerfile#L22